onnxruntime/core/framework/error_code_helper.h

```diff
@@ -8,11 +8,16 @@
 #include "core/session/onnxruntime_c_api.h"
 
 namespace onnxruntime {
+
 // Convert onnxruntime::common::Status to OrtStatus*.
 _Ret_notnull_ OrtStatus* ToOrtStatus(const onnxruntime::common::Status& st);
 
 // Convert OrtStatus* to onnxruntime::common::Status.
+// Note: ToStatus() will not release the OrtStatus*. MoveToStatus() will convert and release the OrtStatus*.
 Status ToStatus(const OrtStatus* ort_status, common::StatusCategory category = common::StatusCategory::ONNXRUNTIME);
+
+// Convert OrtStatus* to onnxruntime::common::Status and release the OrtStatus*.
+Status MoveToStatus(OrtStatus* ort_status, common::StatusCategory category = common::StatusCategory::ONNXRUNTIME);
```

Comment:
I'm open to suggestions for a better name. "move" here is in the sense of C++ move semantics.

Reply:
IMO, the name `ToStatus` implies a conversion and I'd find it unexpected if the source `OrtStatus*` also got released. we're a bit limited in the expressiveness of the signature since we only have a pointer and not some other type where we can use an rvalue reference argument.

in addition to `MoveToStatus`, I also considered `ConvertToStatusAndRelease`. is that any better?

> From what I've seen there's no usage where you want to preserve the OrtStatus if you're converting to onnxruntime::Status.

this is what I've seen too. I can just remove the version that doesn't release it entirely.

Reply:
I find the 'move' slightly misleading as there's a copy of the error message so it's not quite equivalent to a C++ move.

I'd definitely want there to be only one function as I don't think there's a need for the non-releasing version. 

Personally I'd vote for `ToStatusAndRelease`. The `ToStatus` pairs with `ToOrtStatus` so `Convert` seems unnecessary unless you add it to both.

Reply:
ok, renamed it to `ToStatusAndRelease`.